File: connectors-editors/jira-editor/src/main/scala/com/taskadapter/connector/jira/JiraQueryListLoader.scala
package com.taskadapter.connector.jira

import java.util

import com.google.common.base.Strings
import com.taskadapter.connector.definition.WebConnectorSetup
import com.taskadapter.connector.definition.exceptions.{ConnectorException, ServerURLNotSetException}
import com.taskadapter.model.NamedKeyedObject
import com.taskadapter.web.callbacks.DataProvider

class JiraQueryListLoader(jiraConfig: JiraConfig, setup: WebConnectorSetup) extends DataProvider[java.util.List[_ <: NamedKeyedObject]] {

  /**
    * Loads data.
    *
    * @return loaded data.
    * @throws ConnectorException if current state is invalid.
    */
  override def loadData(): util.List[_ <: NamedKeyedObject] = {
    if (Strings.isNullOrEmpty(setup.host)) throw new ServerURLNotSetException
    new JiraConnector(jiraConfig, setup).getFilters
  }
}

File: connectors-editors/jira-editor/src/main/java/com/taskadapter/connector/jira/JiraQueryListLoader.java
package com.taskadapter.connector.jira;

import com.google.common.base.Strings;
import com.taskadapter.connector.definition.WebConnectorSetup;
import com.taskadapter.connector.definition.exceptions.ConnectorException;
import com.taskadapter.connector.definition.exceptions.ServerURLNotSetException;
import com.taskadapter.model.NamedKeyedObject;
import com.taskadapter.web.callbacks.DataProvider;

import java.util.List;

public class JiraQueryListLoader implements DataProvider<List<? extends NamedKeyedObject>> {
    private JiraConfig jiraConfig;
    private WebConnectorSetup setup;

    public JiraQueryListLoader(JiraConfig jiraConfig, WebConnectorSetup setup) {
        this.jiraConfig = jiraConfig;
        this.setup = setup;
    }

    @Override
    public List<? extends NamedKeyedObject> loadData() throws ConnectorException {
        if (Strings.isNullOrEmpty(setup.getHost())) {
            throw new ServerURLNotSetException();
        }
        return new JiraConnector(jiraConfig, setup).getFilters();
    }
}
